nixos/modules/services/networking/pdnsd.nix

```diff
@@ -0,0 +1,93 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.pdnsd;
+  pdnsd = pkgs.lib.overrideDerivation pkgs.pdnsd
+           (attrs:
+             {
+               configureFlags =
+                 [ "--enable-ipv6"
```

Comment:
I think that would be a cleaner solution. I added another two commits to this PR reflecting this update on the default `pdnsd` package.